### DISCUSSION TITLE
Failed to check GitHub API rate limit status.

### DISCUSSION THREAD

**Discussion by lzlcs** [mason-org/mason.nvim#244] [Q&A]
I have checked the `:h mason-errors-github-api`.
Although I have installed GitHub cli, this problem is still unsolved
Here are the details
```

mason: require("mason.health").check()
========================================================================
## mason.nvim report
  - OK: neovim version >= 0.7.0
  - OK: **Go**: `go version go1.18.4 linux/amd64`
  - OK: **cargo**: `cargo 1.62.1 (a748cf5a3 2022-06-08)`
  - WARNING: **luarocks**: unsupported version `/usr/bin/luarocks 2.4.2`. Luarocks version must be >= 3.0.0.
  - OK: **Ruby**: `ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-linux-gnu]`
  - OK: **RubyGem**: `3.1.2`
  - OK: **Composer**: `Composer 1.10.1 2020-03-13 20:34:27`
  - OK: **PHP**: `PHP 7.4.3 (cli) (built: Jun 13 2022 13:43:30) ( NTS )`
  - OK: **npm**: `8.5.5`
  - OK: **node**: `v17.9.0`
  - OK: **python3**: `Python 3.8.10`
  - OK: **pip3**: `pip 20.0.2 from /usr/lib/python3/dist-packages/pip (python 3.8)`
  - WARNING: **javac**: not available
  - WARNING: **java**: not available
  - OK: **julia**: `julia version 1.4.1`
  - OK: **wget**: `GNU Wget 1.20.3 built on linux-gnu.`
  - OK: **curl**: `curl 7.68.0 (x86_64-pc-linux-gnu) libcurl/7.68.0 OpenSSL/1.1.1f zlib/1.2.11 brotli/1.0.7 libidn2/2.2.0 libpsl/0.21.0 (+libidn2/2.2.0) libssh/0.9.3/openssl/zlib nghttp2/1.40.0 librtmp/2.3`
  - OK: **gzip**: `gzip 1.10`
  - OK: **tar**: `tar (GNU tar) 1.30`
  - OK: **bash**: `GNU bash, version 5.0.17(1)-release (x86_64-pc-linux-gnu)`
  - OK: **sh**: `Ok`
  - WARNING: Failed to check GitHub API rate limit status.

```

**Comment by williamboman** [comment]
Is this causing issues with installations or is it just the checkhealth that's not reporting? Can you call the API manually, like so:

```
$ gh api rate_limit
```

**Reply by lzlcs** [reply]
> Is this causing issues with installations or is it just the checkhealth that's not reporting? Can you call the API manually, like so:
> 
> ```
> $ gh api rate_limit
> ```
Thank you for your help first.

After adding the environment variable "GH_TOKEN", 
`Get "https://api.github.com/rate_limit": x509: certificate signed by unknown authority`
I'm going crazy. All kinds of certificate problems bother me, not only here.

**Comment by beauwilliams** [comment]
I have the same issue. `gh` is authenticated. `gh api rate_limit` says over 4900 requests remaining.

Checkhealth also says `  - WARNING: Failed to check GitHub API rate limit status.`

Not sure why... will try figure it out

[M1 mac, homebrew, arm]
